sklearn/linear_model/_cd_fast.pyx

```diff
@@ -273,16 +273,17 @@ def enet_coordinate_descent(
     return np.asarray(w), gap, tol, n_iter + 1
 
 
+# TODO: use const fused typed memoryview where possible when Cython 0.29.33 is used.
```

Comment:
Does this TODO take effect now since const fused typed memoryviews are available?

Reply:
Not yet, but once https://github.com/scikit-learn/scikit-learn/pull/25342#issuecomment-1377262962 is merged.

Reply:
#25342 is now merged. Shall we try that now or is it safer not to experiment with const typed memory views in a bugfix (1.2.1) and instead do that only in `main` (for 1.3.0).

Reply:
Provided that Cython has good coverage and test cases for `const`-qualified memoryviews, I think it is safe and the best to use them to resolve such issues (as it is conceptually the best solution). Yet it do not think we need to use them for this fix with respect to the next bug fix release (I do not want to postpone this release).

Note that using such construct might not be sufficient to fix the problem tackled by this PR (for the reason given in the TODO).

I will start investigating `const`-qualified fused typed memoryviews tomorrow for this fix and for other long standing issues.